File: BoyiaApp/android/app/src/main/kotlin/com/boyia/app/shell/model/BoyiaAppItem.kt
package com.boyia.app.shell.model

import com.boyia.app.common.json.JsonAnnotation.JsonKey

// data class对打印数据友好
data class BoyiaAppItem(
    @field:JsonKey(name = "name") var name: String? = null,
    var cover: String? = null,
    var url: String? = null,
    @field:JsonKey(name = "id") var appId: Int = 0,
    var versionCode: Int = 0
)

File: BoyiaApp/ios/app/app/Model/BoyiaAppItem.swift
//
//  BoyiaAppItem.swift
//  app
//
//  Created by yanbo on 2022/4/27.
//

import Foundation

// Identifiable用于list遍历，Codable用于json解析
struct BoyiaAppItem: Codable, Identifiable {
    var id = UUID()
    let appId: Int
    let name: String
    let cover: String
    let url: String
    let versionCode: Int
    
    // 映射json字段名
    enum CodingKeys: String, CodingKey {
        case appId = "id"
        case name
        case cover
        case url
        case versionCode
    }
}
